src/core/operations/YaraRules.mjs

```diff
@@ -0,0 +1,82 @@
+/**
+ * @author Matt C [matt@artemisbot.uk]
+ * @copyright Crown Copyright 2019
+ * @license Apache-2.0
+ */
+
+import Operation from "../Operation";
+import OperationError from "../errors/OperationError";
+import Yara from "libyara-wasm";
+
+/**
+ * Yara Rules operation
+ */
+class YaraRules extends Operation {
+
+    /**
+     * YaraRules constructor
+     */
+    constructor() {
+        super();
+
+        this.name = "Yara Rules";
+        this.module = "Yara";
+        this.description = "Yara support";
+        this.infoURL = "https://en.wikipedia.org/wiki/YARA";
+        this.inputType = "ArrayBuffer";
+        this.outputType = "string";
+        this.args = [{
+            name: "Rules",
+            type: "code",
+            value: ""
+        }];
+    }
+
+    /**
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {string}
+     */
+    run(input, args) {
+        return new Promise((resolve, reject) => {
+            Yara().then(yara => {
+                let matchString = "";
+                const inpArr = new Uint8Array(input); // I know this is garbage but it's like 1.5 times faster
```

Comment:
Why is this garbage? `input` is an ArrayBuffer, so this is the correct way to view it as a Uint8Array.

Reply:
Oh yes, typed arrays are way more efficient, especially when working with large files. If you use a byte array, the whole thing has to be converted between types multiple times. Always use `ArrayBuffer` over `byteArray` if you expect the input to be a file.